src/pyodide/internal/workers.py

```diff
@@ -149,54 +157,202 @@ def json(
             raise _to_python_exception(exc) from exc
 
 
-# TODO: Implement pythonic blob API
-FormDataValue = "str | js.Blob"
+FormDataValue = "str | js.Blob | Blob"
 
 
 class FormData(MutableMapping[str, FormDataValue]):
-    def __init__(self, form_data: "js.FormData | None | dict[str, FormDataValue]"):
+    """
+    This API follows that of https://pypi.org/project/multidict/.
+    """
+
+    def __init__(
+        self, form_data: "js.FormData | None | dict[str, FormDataValue]" = None
+    ):
         if form_data:
             if isinstance(form_data, dict):
-                self.js_form_data = js.FormData.new()
+                self._js_form_data = js.FormData.new()
                 for item in form_data.items():
-                    self.js_form_data.append(item[0], item[1])
+                    self._js_form_data.append(
+                        item[0], self._process_value_for_js(item[1])
+                    )
             else:
-                self.js_form_data = form_data
+                self._js_form_data = form_data
+        else:
+            self._js_form_data = js.FormData.new()
+
+    def _process_value_for_js(self, item: FormDataValue) -> "str | js.Blob":
+        if isinstance(item, Blob):
+            return item.native
         else:
-            self.js_form_data = js.FormData.new()
+            return item
 
-    def __getitem__(self, key: str) -> list[FormDataValue]:
-        return list(self.js_form_data.getAll(key))
+    def _process_value_for_py(self, item: FormDataValue) -> "str | Blob":
+        if hasattr(item, "constructor") and (item.constructor.name in ("Blob", "File")):
+            return Blob(item)
+        else:
+            return item
```

Comment:
nits: These also could be top level nonstatic functions. Maybe call them `_js_value_to_py` and `_py_value_to_js`?